dynomite-derive/src/lib.rs

```diff
@@ -54,33 +52,52 @@ struct ItemField<'a> {
 impl<'a> ItemField<'a> {
     fn new(field: &'a Field) -> Self {
         let attrs = parse_attrs(&field.attrs);
-        Self { field, attrs }
+        let me = Self { field, attrs };
+        if me.is_flatten() {
+            if let Some(it) = me
```

Comment:
Let's try to make this contract clear in the docs to avoid surprising users

Reply:
I've added more docs for the attributes to the crate level of `dynomite` (this is basically the entrypoint for our API discovery).
See commit 5e36924.

I'd like to merge this without too many more additions not to bloat this PR, I am inclined to make a cleanup in a separate PR. As you've mentioned it is better to do atomic PRs with targeted changes